Client/Frontend/Browser/Playlist/Utilities/PlaylistStatusObserver.swift

```diff
@@ -17,40 +17,33 @@ class PlaylistPlayerStatusObserver: NSObject {
   private var item: AVPlayerItem?
   private var onStatusChanged: (AVPlayerItem.Status) -> Void
   private var currentItemObserver: NSKeyValueObservation?
-  private var itemStatusObserver: NSKeyValueObservation?
 
   init(player: AVPlayer, onStatusChanged: @escaping (AVPlayerItem.Status) -> Void) {
     self.onStatusChanged = onStatusChanged
     super.init()
 
     self.player = player
     currentItemObserver = player.observe(
-      \AVPlayer.currentItem, options: [.new],
+      \AVPlayer.currentItem?.status, options: [.new],
       changeHandler: { [weak self] _, change in
         guard let self = self else { return }
 
-        if let newItem = change.newValue {
-          self.item = newItem
-          self.itemStatusObserver = newItem?.observe(
-            \AVPlayerItem.status, options: [.new],
-            changeHandler: { [weak self] _, change in
-              guard let self = self else { return }
-
-              let status = change.newValue ?? .unknown
-              switch status {
-              case .readyToPlay:
-                log.debug("Player Item Status: Ready")
-                self.onStatusChanged(.readyToPlay)
-              case .failed:
-                log.debug("Player Item Status: Failed")
-                self.onStatusChanged(.failed)
-              case .unknown:
-                log.debug("Player Item Status: Unknown")
-                self.onStatusChanged(.unknown)
-              @unknown default:
-                assertionFailure("Unknown Switch Case for AVPlayerItemStatus")
-              }
-            })
+        let status = change.newValue ?? .unknown
+        switch status {
+        case .readyToPlay:
+          log.debug("Player Item Status: Ready")
+          self.onStatusChanged(.readyToPlay)
+        case .failed:
+          log.debug("Player Item Status: Failed")
+          self.onStatusChanged(.failed)
+        case .unknown:
```

Comment:
you can also remove the `?? .unknown` above and add a `.none` case

Reply:
No you can't because `change.newValue` is a double optional.